specs/trip_dispatch_spec.rb

```diff
@@ -111,4 +95,67 @@
       expect(passenger.trips).must_include trip
     end
   end
+
+  describe '#request_trip method' do
+    before do
+      @requested_trip = @dispatcher.request_trip(1)
+    end
+
+    it "generates an array of :AVAILABLE drivers" do
+      available_drivers = @dispatcher.generate_available_drivers
+      expect(available_drivers).must_be_kind_of Array
+      expect(available_drivers.first.status).must_equal :AVAILABLE
+      expect(available_drivers.last.status).must_equal :AVAILABLE
```

Comment:
I would go further than just checking the first and the last here: iterate through the list and check each one.